lib/BasicAnalyses/RemoveHelperCalls.cpp

```diff
@@ -0,0 +1,85 @@
+/// \file RemoveHelperCalls.cpp
+/// \brief Remove calls to helpers in a function and replaces them
+///        with stores of an opaque value onto the CSVs clobbered by the helper.
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/SmallVector.h"
+
+#include "revng/BasicAnalyses/RemoveHelperCalls.h"
+#include "revng/Support/IRHelpers.h"
+
+llvm::PreservedAnalyses
+RemoveHelperCallsPass::run(llvm::Function &F,
+                           llvm::FunctionAnalysisManager &FAM) {
+  using namespace llvm;
+
+  // Get the result of the GCBI analysis
+  auto &M = *F.getParent();
+  const auto &MAMProxy = FAM.getResult<ModuleAnalysisManagerFunctionProxy>(F);
+  GCBI = MAMProxy.getCachedResult<GeneratedCodeBasicInfoAnalysis>(M);
+  if (!GCBI)
+    GCBI = &(FAM.getResult<GeneratedCodeBasicInfoAnalysis>(F));
+  revng_assert(GCBI != nullptr);
```

Comment:
Is this invoked both from module- and function-pass managers?

Reply:
Forgot to remove the module-pass related stuff from here too. Fixed.